network-config-analyzer/TopologyObjectsFinder.py

```diff
@@ -102,8 +102,10 @@ def _add_pod_from_yaml(self, pod_object):
 
         containers = pod_object['spec'].get('containers', {})
         for container in containers:
-            for port in container.get('ports', {}):
-                pod.add_named_port(port.get('name'), port.get('containerPort'), port.get('protocol', 'TCP'))
+            ports = container.get('ports', {})
+            if ports:
+                for port in ports:
+                    pod.add_named_port(port.get('name'), port.get('containerPort'), port.get('protocol', 'TCP'))
```

Comment:
```suggestion
            for port in container.get('ports') or  []:
                pod.add_named_port(port.get('name'), port.get('containerPort'), port.get('protocol', 'TCP'))
```
Avoiding extra indentation. Please also apply to the rest of the changes